### ISSUE TITLE
False positives in update-alternatives check when multiple subpackages use it

### ISSUE THREAD

**Issue by aaronpuchert** [rpm-software-management/rpmlint#835]
There are errors like this:
```
lld13.x86_64: E: update-alternatives-postun-call-missing clang
lldb13.x86_64: E: update-alternatives-postun-call-missing clang
lldb13.x86_64: E: update-alternatives-postun-call-missing lld
lldb13.x86_64: E: update-alternatives-postun-call-missing ld
llvm13.x86_64: E: update-alternatives-postun-call-missing clang
llvm13.x86_64: E: update-alternatives-postun-call-missing lld
llvm13.x86_64: E: update-alternatives-postun-call-missing ld
llvm13.x86_64: E: update-alternatives-postun-call-missing lldb
The package does not call update-alternatives --remove in postun phase to
remove all the configuration for each individual --install binary that was
done in postun.
```
and later
```
lld13.x86_64: E: alternative-link-missing /etc/alternatives/clangd
lld13.x86_64: E: alternative-link-missing /etc/alternatives/c-index-test
lld13.x86_64: E: alternative-link-missing /etc/alternatives/clang++
[...]
lldb13.x86_64: E: alternative-link-missing /etc/alternatives/clangd
lldb13.x86_64: E: alternative-link-missing /etc/alternatives/c-index-test
lldb13.x86_64: E: alternative-link-missing /etc/alternatives/clang++
[...]
llvm13.x86_64: E: alternative-link-missing /etc/alternatives/clangd
llvm13.x86_64: E: alternative-link-missing /etc/alternatives/c-index-test
llvm13.x86_64: E: alternative-link-missing /etc/alternatives/clang++
[...]
The file %{_sysconfdir}/alternatives/$(basename generic-name) is missing in
the file list. Mark it as %ghost and add it to the file list.

lld13.x86_64: E: alternative-generic-name-missing /usr/bin/clangd
[...]
The update-alternatives generic name is not in the filelist. Create it as a
symlink to %{_sysconfdir}/alternatives/$(basename generic-name) and add it to
the file list.
```

These seem like false positives to me. There are four subpackages using update-alternatives, and the check seems to not consider which files belong in which package, e.g. all the clang binaries are in `clang13`, not in `lld13`, `lldb13` or  `llvm13`, so there is no need for update-alternatives handling related to clang binaries in the other packages.

Looking at the errors for `update-alternatives-postun-call-missing` it seems that the check runs on the packages in the some order, because we're not seeing an error for `clang13`, then an error for `lld13` with regards to `clang13`, then errors for `lldb13` about the `clang13` and `lld13` and lastly `llvm13` for all other packages.

For reference, the package is at https://build.opensuse.org/package/show/devel:tools:compiler/llvm13.

**Comment by marxin** [comment]
Confirmed, hope I've got a fix for it.
